lib/person.rb

```diff
@@ -0,0 +1,25 @@
+class Person
+  attr_reader :name,
+              :interests,
+              :supplies
+  def initialize(data)
+    @name       = data[:name]
+    @interests  = data[:interests]
+    @supplies   = Hash.new(0)
+  end
+
+  def add_supply(material, amount)
+    @supplies[material] += amount
+  end
+
+
+  def can_build?(craft)
+      supply = @supplies.transform_keys {|k,v| k.to_sym}
+      if (supply.keys & craft.supplies_required.keys).all? && supply.values.sum > craft.supplies_required.values.sum
+        true
+      else
+        false
+      end
+    end
```

Comment:
I don't think logic here is quite right. You need to check the quantity of each supply individually, rather than summing all the quantities from both the person's supplies and the craft's supplies. Also, be careful with using the `&` operator. I think what you meant to use was the `&&` operator, but they do not do the same thing.